torch/_inductor/codegen/common.py

```diff
@@ -39,13 +39,25 @@ def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
 class ExprPrinter(Printer):
     @staticmethod
     def paren(string):
+        def matching_parens(string):
+            op = []
+            dc = {
+                op.pop() if op else -1: i
+                for i, c in enumerate(string)
+                if (c == "(" and op.append(i) and False) or (c == ")" and op)
+            }
+            return False if dc.get(-1) or op else dc
```

Comment:
This looks slow

Reply:
Replaced with a simpler version with an early exit. W/o early exit it's still ~4 us for ~50 symbols expr, early exit is faster (depending on where it happened)